### PR TITLE
ILM stop step execution if writeIndex is false

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

**Review by AthenaEryma — CHANGES_REQUESTED** [review]
Nice catch, thanks for fixing. That said, I have a few issues with the tests.

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStepTests.java [hunk 1]
```diff
@@ -209,6 +213,64 @@ public void onFailure(Exception e) {
         Mockito.verify(indicesClient, Mockito.only()).rolloverIndex(Mockito.any(), Mockito.any());
     }
 
+    public void testPerformActionWriteIndexIsFalse() {
```
[the remark this answers is not in the document]
Comment (andreidan) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStepTests.java:216]: thanks for the suggestion, adding a verify on the client mock

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStepTests.java [hunk 2]
```diff
@@ -209,6 +213,64 @@ public void onFailure(Exception e) {
...
+
+            @Override
+            public void onFailure(Exception e) {
+                assertThat(e.getMessage(), is(String.format(Locale.ROOT, "index [%s] is not the write index for alias [%s]",
+                    indexMetadata.getIndex().getName(), alias)));
+            }
+        }, MASTER_TIMEOUT);
+    }
+
+    public void testPerformActionWriteIndexIsNullOrTrue() {
```
[the remark this answers is not in the document]
Comment (andreidan) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStepTests.java:240]: good shout! (I like to see unit tests as specification, collapse them and just by reading the test names understand what the invariants are, but in this case I think it makes sense to drop the extra test) Thanks for the suggestion

**Comment by andreidan** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by dakrone — APPROVED** [review]
LGTM

**Pull request merged**